pkg/metrics/prometheus_test.go

```diff
@@ -16,212 +16,90 @@
 package metrics
 
 import (
-	"fmt"
+	"context"
 	"net/http"
-	"net/http/httputil"
-	"reflect"
+	"net/http/httptest"
 	"strings"
 	"testing"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/client_golang/prometheus/testutil"
 )
 
-func TestNewPrometheus(t *testing.T) {
-	test := struct {
-		name string
-		want Metrics
-	}{
-		name: "default",
-		want: &prom{
-			counters: make(map[string]prometheus.Counter),
-			summary:  make(map[string]prometheus.Summary),
-		},
+func TestMetrics(t *testing.T) {
```

Comment:
this test function is testing multiple things at the same time, can we split this up to test smaller subsections of the metrics module?

Reply:
👍